src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3554,6 +3554,36 @@ std::optional<UInt64> StorageReplicatedMergeTree::totalRows() const
     return res;
 }
 
+std::optional<UInt64> StorageReplicatedMergeTree::totalRowsByPartitionPredicate(const SelectQueryInfo & query_info, const Context & context) const
```

Comment:
It's 100% copy-paste, need to move common code to `MergeTreeData`.

Reply:
I didn't find a good way to abstract, and is worry about introducing non-trivial template code.